### PR TITLE
Adding s3 bucket sink

### PR DESCRIPTION
**Pull request by matzew**
# Changes

- as per title

/kind <kind>

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: aws-s3-sink/pom.xml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      <artifactId>citrus-quarkus</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.citrusframework</groupId>
+      <artifactId>citrus-http</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty.ee10</groupId>
```
Comment (christophd) [inline aws-s3-sink/pom.xml:123]: The dependency is not required anymore as we have pinned the version in dependencyManagement in parent POM

File: aws-s3-sink/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+quarkus.camel.openapi.codegen.model-package=org.apache.camel.quarkus
+
+quarkus.native.resources.includes=camel/kn-connector-source.yaml
+camel.main.routes-include-pattern=camel/kn-connector-source.yaml
+
+# Knative Camel component
+camel.component.knative.environmentPath=classpath:knative.json
+
+camel.component.knative.ceOverride[ce-type]={{kn.connector.ce.override.type:dev.knative.connector.event.aws-s3}}
```
Comment (christophd) [inline aws-s3-sink/src/main/resources/application.properties:26]: these ceOverrides are not required for the sink

File: aws-s3-sink/src/main/resources/camel/kn-connector-source.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ---------------------------------------------------------------------------
+
+apiVersion: camel.apache.org/v1
+kind: Pipe
+metadata:
+  name: kn-connector-aws-s3-sink
+spec:
+  source:
```
Comment (christophd) [inline aws-s3-sink/src/main/resources/camel/kn-connector-source.yaml:23]: source should be the Knative Broker

File: aws-s3-sink/src/main/resources/camel/kn-connector-source.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    ref:
+      apiVersion: camel.apache.org/v1
+      kind: Kamelet
+      name: aws-s3-sink
+  sink:
```
Comment (christophd) [inline aws-s3-sink/src/main/resources/camel/kn-connector-source.yaml:28]: sink should be the aws-s3-sink Kamelet

**Review by christophd — COMMENTED** [review]
The sink-connector has mixed a few things coming from a source-connector (e.g. in kn-connector-source.yaml which should rather follow the concepts of kn-connector-sink.yaml). 

Please have a look into log-sink connector as an example or use the Maven archetype for sink connectors to generate the project

```
mvn archetype:generate \
 -DarchetypeGroupId=dev.knative.eventing \
 -DarchetypeArtifactId=kn-connector-archetype-sink \
 -DarchetypeVersion=1.0-SNAPSHOT
```

**Comment by matzew** [comment]
I used the archetype now as:

```
mvn archetype:generate \
 -DarchetypeGroupId=dev.knative.eventing \
 -DarchetypeArtifactId=kn-connector-archetype-sink \
 -DarchetypeVersion=1.0-SNAPSHOT \
 -DartifactId=aws-s3-sink \
 -Dkamelet-name=aws-s3 \
 -Dcomponent-name=aws2-s3
```

the `pom.xml` diff is manually - I just added the sink module.

I reverted the change from the archetype since it was adding new lines after each existing line. Was very verbose

**Comment by christophd** [comment]
> I reverted the change from the archetype since it was adding new lines after each existing line. Was very verbose
yes, Maven archetype process goes wild on formatting the existing POM. Revert and adding the module manually is the best option for now. thanks

File: aws-s3-sink/properties.adoc [hunk 5]
```diff
@@ -0,0 +1,4 @@
+|===
+|Property |Required |EnvVar |Description
```
Comment (christophd) [inline aws-s3-sink/properties.adoc:2]: consider running `mvn package` once before the PR, so the docs get generated

**Review by christophd — APPROVED** [review]
LGTM now

**Pull request merged**